Alluvial/StreamCatchup{T}.cs

```diff
@@ -75,62 +75,44 @@ internal class StreamCatchup<TData> : IStreamCatchup<TData>
 
-                                    if (batch.Count > 0)
-                                    {
-                                        var aggregatorUpdates = aggregatorSubscriptions
-                                            .Values
-                                            .Select(subscription =>
-                                            {
-                                                try
-                                                {
-                                                    return Aggregate(s.Id,
-                                                                     (dynamic) subscription,
-                                                                     batch,
-                                                                     streamQuery.Cursor);
-                                                }
-                                                catch (RuntimeBinderException exception)
-                                                {
-                                                    throw new RuntimeBinderException(
-                                                        string.Format("Unable to call Aggregate, possibly due to {0} using non-public types for its generic parameters",
-                                                                      subscription), exception);
-                                                }
-                                            })
-                                            .Cast<Task>();
+                            var batch = await streamQuery.NextBatch();
 
+                            if (batch.Count > 0)
+                            {
+                                var aggregatorUpdates = aggregatorSubscriptions
+                                    .Values
+                                    .Select(subscription =>
+                                    {
                                         try
                                         {
-                                            await Task.WhenAll(aggregatorUpdates);
+                                            return Aggregate(s.Id,
+                                                             (dynamic) subscription,
+                                                             batch,
+                                                             streamQuery.Cursor);
                                         }
-                                        catch (Exception exception)
+                                        catch (RuntimeBinderException exception)
                                         {
-                                            aggregatorException = exception;
-                                            throw;
+                                            throw new RuntimeBinderException(
+                                                string.Format("Unable to call Aggregate, possibly due to {0} having non-public types for its generic parameters",

```

Comment:
This is an implementation detail that we can fix by using expression trees rather than dynamic dispatch but for now it's not critical. 



> On Jan 31, 2015, at 2:51 PM, Nathan Black <notifications@github.com> wrote:
> 
> In Alluvial/StreamCatchup{T}.cs:
> 
> >                                          {
> > -                                            aggregatorException = exception;
> > -                                            throw;
> > +                                            throw new RuntimeBinderException(
> > +                                                string.Format("Unable to call Aggregate, possibly due to {0} having non-public types for its generic parameters",
> hehe +1
> 
> —
> Reply to this email directly or view it on GitHub.
>